pkg/session/redis/sess_redis.go

```diff
@@ -212,38 +206,34 @@ func (rp *Provider) SessionRead(sid string) (session.Store, error) {
 
 // SessionExist check redis session exist by sid
 func (rp *Provider) SessionExist(sid string) (bool, error) {
-	c := rp.poollist.Get()
-	defer c.Close()
+	c := rp.poollist
 
-	if existed, err := redis.Int(c.Do("EXISTS", sid)); err != nil || existed == 0 {
+	if existed, err := c.Exists(sid).Result(); err != nil || existed == 0 {
 		return false, err
 	}
 	return true, nil
 }
 
 // SessionRegenerate generate new sid for redis session
 func (rp *Provider) SessionRegenerate(oldsid, sid string) (session.Store, error) {
-	c := rp.poollist.Get()
-	defer c.Close()
-
-	if existed, _ := redis.Int(c.Do("EXISTS", oldsid)); existed == 0 {
+	c := rp.poollist
+	if existed, _ := c.Exists(oldsid).Result(); existed == 0 {
 		// oldsid doesn't exists, set the new sid directly
 		// ignore error here, since if it return error
 		// the existed value will be 0
-		c.Do("SET", sid, "", "EX", rp.maxlifetime)
+		c.Do(c.Context(), "SET", sid, "", "EX", rp.maxlifetime)
 	} else {
-		c.Do("RENAME", oldsid, sid)
-		c.Do("EXPIRE", sid, rp.maxlifetime)
+		c.Rename(oldsid, sid)
+		c.Expire(sid, time.Duration(rp.maxlifetime))
```

Comment:
It seems that `Expire key1 10` means that the key1 will be expired after 10 **SECOND**? so if we use `time.Duration(rp.maxlifetime)`, it will be a problem.

Reply:
MaxLifetime is set to 3600 by default.
The time was being provided in seconds before as well so I think this should be fine.

I am not really sure why the sid gets expired in the first place though.

Reply:
But there is a `* time.Second` missing there

Reply:
Okay understood the Expiry. Because redis does not has a garbage collection defined, all entries get an expiry time.